google_search_tool.py

```diff
@@ -0,0 +1,123 @@
+from langgraph.graph import StateGraph, END
+from langchain_core.runnables import RunnableLambda
+from langchain_core.tools import tool
+from langchain_core.messages import HumanMessage
+from langchain_openai import ChatOpenAI
+from serpapi import GoogleSearch
+from typing import TypedDict, List
+from dotenv import load_dotenv
+import os
+import re
+
+load_dotenv()
+SERPAPI_API_KEY = os.getenv("serp_api_key")
+OPENAI_API_KEY = os.getenv("openai_api_key")
+
+# -------------------------
+# Tool: Google Search via SerpAPI
+# -------------------------
+@tool
+def google_search_tool(query: str) -> dict:
+    """Search Google and return the top 3 clean text results."""
+    search = GoogleSearch({
+        "q": query,
+        "api_key": SERPAPI_API_KEY,
+        "num": 3
+    })
+    results = search.get_dict()
+    output = []
+
+    for result in results.get("organic_results", [])[:3]:
+        content = result.get("snippet") or ""
+        clean_text = re.sub(r'<.*?>', '', content)
+        output.append({
+            "url": result.get("link", ""),
+            "content": clean_text.strip()
+        })
+
+    return {
+        "answer": f"Here are the top 3 results for: {query}",
+        "results": output
+    }
+
+# -------------------------
+# Typed Structures
+# -------------------------
+class SearchResult(TypedDict):
+    url: str
+    content: str
+
+class SearchOutput(TypedDict):
+    answer: str
+    results: List[SearchResult]
+
+class AgentState(TypedDict):
+    input: str
+    results: SearchOutput
+    final_answer: str
```

Comment:
The agent state should have the same structure as the calendar agent. it is needed for the orchestrator agent that I am actually working on

```suggestion
class WorkersState(TypedDict):
    """The state of the worker agents."""
    workers_messages: Annotated[Sequence[BaseMessage], add_messages]
    user_id: str
    account_name: str
    num_calls: Annotated[list[str], add]

```

Reply:
updated the structure of the google search agent with same as the calender agent.